oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/provider/parser/listener/endpoint/MultiScopesSpanListener.java

```diff
@@ -179,13 +187,13 @@ public void parseEntry(SpanDecorator spanDecorator, SegmentCoreInfo segmentCoreI
             for (KeyStringValuePair tag : spanDecorator.getAllTags()) {
                 if (SpanTags.DB_STATEMENT.equals(tag.getKey())) {
                     String sqlStatement = tag.getValue();
-                    if (!StringUtil.isEmpty(sqlStatement) && sqlStatement.length() > config.getMaxSlowSQLLength()) {
-                        statement.setStatement(sqlStatement.substring(0,config.getMaxSlowSQLLength()));
-                    }
-                    else {
+                    if (StringUtil.isEmpty(sqlStatement)) {
+                        statement.setStatement("No statement.");
```

Comment:
If no statement, how about using operation name?
Such as `statement=[No statement]/commit`

Reply:
you mean `statement.setStatement(spanDecorator.getOperationName());`?

Reply:
Yes, logically, but you may need ID-NAME exchange? As agent could upload the ID only?  Such as this
```java
String endpointName = spanObject.getOperationName();
            if (spanObject.getOperationNameId() != 0) {
                EndpointInventory endpointInventory = getEndpointInventoryCache().get(spanObject.getOperationNameId());
                if (nonNull(endpointInventory)) {
                    endpointName = endpointInventory.getName();
                } else {
                    endpointName = Const.EMPTY_STRING;
                }
            }
            span.setEndpointName(endpointName);
```

Just a thought, because I think the pure `No statement` is not much more helper than blank. What do you think?

Reply:
Agree, I will add more detail in statement.